config/config_test.go

```diff
@@ -557,3 +900,117 @@ func TestBreakUpConfig_Conflict_Value(t *testing.T) {
 	got := input.breakUp()
 	is.Equal(want, got)
 }
+
+func TestConfig_getValuesForParameter(t *testing.T) {
+	cfg := Config{
+		"ignore": "me",
+
+		// foo
+		"test.foo.val": "0",
+
+		"test.foo.format.baz.type":    "0",
+		"test.foo.format.baz.options": "0",
+
+		"test.foo.format.qux.type":    "0",
+		"test.foo.format.qux.options": "0",
+
+		// bar
+		"test.bar.val": "0",
+
+		"test.bar.format.baz.type":    "0",
+		"test.bar.format.baz.options": "0",
+
+		"test.bar.format.qux.type":    "0",
+		"test.bar.format.qux.options": "0",
+
+		// include
+		"test.include.me": "yes",
+
+		// ignore this, it's not nested
+		"test.ignore": "0",
+	}
+
+	testCases := []struct {
+		key  string
+		want []string
+	}{{
+		key:  "test.foo.val",
+		want: []string{"test.foo.val"},
+	}, {
+		key:  "blah",
+		want: []string{"blah"},
+	}, {
+		key: "test.*.blah",
+		want: []string{
+			// Note that the function returns keys that don't exist in the config,
+			// it figures out the potential keys based on matched wildcards.
+			// However, it does not return test.ignore.blah, as test.ignore does
+			// not contain any nested keys.
+			"test.foo.blah",
+			"test.bar.blah",
+			"test.include.blah",
+		},
+	}, {
+		key: "test.*",
+		want: []string{
+			"test.foo.val",
+			"test.foo.format.baz.type",
+			"test.foo.format.baz.options",
+			"test.foo.format.qux.type",
+			"test.foo.format.qux.options",
+			"test.bar.val",
+			"test.bar.format.baz.type",
+			"test.bar.format.baz.options",
+			"test.bar.format.qux.type",
+			"test.bar.format.qux.options",
+			"test.include.me",
+			"test.ignore",
+		},
+	}, {
+		key: "test.*.val",
+		want: []string{
+			"test.foo.val",
+			"test.bar.val",
+			"test.include.val",
+		},
+	}, {
+		key: "test.*.format.*",
+		want: []string{
+			"test.foo.format.baz.type",
+			"test.foo.format.baz.options",
+			"test.foo.format.qux.type",
+			"test.foo.format.qux.options",
+			"test.bar.format.baz.type",
+			"test.bar.format.baz.options",
+			"test.bar.format.qux.type",
+			"test.bar.format.qux.options",
+		},
+	}, {
+		key: "test.*.format.*.type",
+		want: []string{
+			"test.foo.format.baz.type",
+			"test.foo.format.qux.type",
+			"test.bar.format.baz.type",
+			"test.bar.format.qux.type",
+		},
+	}, {
+		key: "test.*.format.*.options",
```

Comment:
what case could need two wildcard characters? and how would this be parsed into a map? should we just have a constrain of 1 wildcard per key?